src/test/java/com/eighthlight/fabrial/test/server/TcpServerConnectionIntegrationTest.java

```diff
@@ -0,0 +1,44 @@
+package com.eighthlight.fabrial.test.server;
+
+import com.eighthlight.fabrial.test.client.TcpClient;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+
+import static com.github.grantwest.eventually.EventuallyLambdaMatcher.eventuallyEval;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+public class TcpServerConnectionIntegrationTest extends TcpServerIntegrationTest {
```

Comment:
The test inheritance here makes it hard to see dependencies. Let's use [composition over inheritance](https://en.wikipedia.org/wiki/Composition_over_inheritance) if you need code sharing - otherwise I'd prefer duplication across code files to test class inheritance. [This article] has a few more arguments if you're interested.

I think this specific issue will go away when you move to unit testing, but it's good to keep in mind anyway.

Reply:
Right, in python I would have simply factored out the client & sever into reusable fixtures. Instead, I went the lazy route and lifted them into a superclass. I believe Junit has a better way...

Reply:
I guess I shouldn't be surprised, but creating reusable fixtures (i.e. Extensions) seems [fairly involved](https://www.baeldung.com/junit-5-extensions). Do you know a better way?

Reply:
I tend to extract normal methods to a helper/support class when needed, and I’m OK with similar setup in tests. I try to remind myself that when I want the same test code multiple places, that’s an opportunity to introduce a seam (if needed) to push the tests down a level and limit the number of tests that need that higher-level setup.